pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/NoDictionarySingleColumnGroupKeyGenerator.java

```diff
@@ -369,6 +388,146 @@ public boolean hasNext() {
 
     @Override
     public GroupKey next() {
+      Object2IntMap.Entry entry = _iterator.next();
+      _groupKey._groupId = entry.getIntValue();
+      _groupKey._keys = new Object[]{entry.getKey()};
+      return _groupKey;
+    }
+
+    @Override
+    public void remove() {
+      throw new UnsupportedOperationException();
+    }
+  }
+
+  private static class IntStringGroupKeyIterator implements Iterator<StringGroupKey> {
```

Comment:
I didnt quite follow why PQL still needs the StringGroupKey?

Reply:
PQL engine is using string as the key to group the results (it does not require sorting on per-group-by column basis, but always sort on aggregation values). We can remove these classes once deprecating PQL.